src/Giraffe/Preconditional.fs

```diff
@@ -88,7 +89,8 @@ type HttpContext with
             match lastModified with
             | None              -> AllConditionsMet
             | Some lastModified ->
-                match  requestHeaders.IfModifiedSince.Value <= DateTimeOffset.UtcNow
+                let lastModified = lastModified.CutOffMs()
+                match  requestHeaders.IfModifiedSince.Value <= DateTimeOffset.UtcNow.CutOffMs()
```

Comment:
Might need an integration test to check the value of `requestHeaders.IfModifiedSince.Value`